.github/workflows/build.yaml

```diff
@@ -8,9 +8,62 @@
 name: Build FPM Images
 on:
   workflow_dispatch:
+    inputs:
+      version:
+        description: The version to use in the image label
+        required: true
+        type: string
+  push:
+    tags: [ '*' ]
+
 jobs:
-  dummy:
+  build:
+    strategy:
+      matrix:
+        include:
+          - directory: centos/8
+            label: centos8
+          - directory: debian/12
+            label: debian12
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write  # required for aws-actions/configure-aws-credentials
+      contents: read
     steps:
-      - name: Greeting
-        run: echo "This is a dummy build to placate GitHub."
+      - name: version
+        env:
+          VERSION: ${{ inputs.version }}
+          REF_NAME: ${{ github.ref_name }}
+        run: |
+          V="${VERSION:-${REF_NAME}}"
```

Comment:
I'm not sure `REF_NAME` is safe.  I think git allows stuff like `/` or `+` that docker registries will choke on.

Probably not a problem, as I don't expect these to show up in tag names even if they're technically legal.

Reply:
I've fixed this to replace all invalid chars with hyphens.